pkgs/by-name/op/opcua-commander/package.nix

```
@@ -0,0 +1,37 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, typescript
+, esbuild
+, makeWrapper
+, nodejs
+}:
+buildNpmPackage rec {
+  pname = "opcua-commander";
+  version = "0.37.0";
+
+  src = fetchFromGitHub {
+    owner = "node-opcua";
+    repo = "opcua-commander";
+    rev = version;
+    hash = "sha256-wQXSSNinY85Ti+D/zklYP2N8IP3OsN9xQNJuuQr4kVU=";
+  };
+
+  npmDepsHash = "sha256-Ux1X/3sam9WHrTfqoWv1r9p3pJOs6BaeFsxHizAvjXA=";
+  nativeBuildInputs = [ esbuild typescript makeWrapper ];
+
+  buildPhase = "tsc -b && npx -y ${esbuild}/bin/esbuild --bundle --outfile=bin/bundle.js --external:blessed  --external:chalk --external:wrap-ansi --minify-syntax --sourcemap --platform=node lib/index.ts";
```

A bit more readable:
```suggestion
  buildPhase = ''
    tsc -b && npx -y ${esbuild}/bin/esbuild \
      --bundle \
      --outfile=bin/bundle.js \
      --external:blessed  \
      --external:chalk \
      --external:wrap-ansi \
      --minify-syntax \
      --sourcemap \
      --platform=node \
      lib/index.ts
  '';
```